### ISSUE TITLE
file upload fails if using the latest crow_all.h

### ISSUE THREAD

**Issue by mochechan** [CrowCpp/Crow#558]
This is my source code. 
```cpp
#define CROW_MAIN  // let the compiler know this is your main cpp file
#define CROW_USE_LOCALTIMEZONE 1
#include "../crow_all.h"
//#include "/ssd/crow_all.h"
#include "../json.hpp"

class ExampleLogHandler : public crow::ILogHandler {
    public:
        void log(std::string /*message*/, crow::LogLevel /*level*/) override {
//            cerr << "ExampleLogHandler -> " << message;
        }
};

struct ExampleMiddleware 
{
    std::string message;

    ExampleMiddleware() : message("foo") {
    }

    void setMessage(const std::string &newMsg) {
        message = newMsg;
    }

    struct context { };

    void before_handle(crow::request& req, crow::response& res, context& ctx) {
        CROW_LOG_DEBUG << " before - MESSAGE: " << message;
    }

    void after_handle(crow::request& req, crow::response& res, context& ctx) {
        CROW_LOG_DEBUG << " after - MESSAGE: " << message;
    }
};


int main() {
	//crow::App<ExampleMiddleware> app;
  //app.get_middleware<ExampleMiddleware>().setMessage("hello");

  crow::SimpleApp app;                 // define your crow application
//  app.loglevel(crow::LogLevel::Info);  
 app.loglevel(crow::LogLevel::Debug);
	//crow::logger::setHandler(std::make_shared<ExampleLogHandler>());

  CROW_ROUTE(app, "/uploadFiles").methods(crow::HTTPMethod::POST)([&]( const crow::request& req) {
    crow::multipart::message msg(req);
	std::string aa = req.get_header_value("aa");

	std::cout << "===================== enumerate files/headers\n";
    for (int i = 0; i < msg.parts.size(); ++i) {
		std::string name, filename, fileContent = msg.parts[i].body;
       	for (auto it=msg.parts[i].headers.begin(); it != msg.parts[i].headers.end();++it) {
			auto const &key = it->first;
			auto const &value = it->second;
			std::cout << " " << key << ":" << value.value << "\n";
			for(const auto &[k,v]: value.params){
				std::cout << "k:"<< k << " v:" << v << "\n";
				if (k.compare("filename") == 0){
            	    filename = v;
	              } else if (k.compare("name") == 0){
    	            name = v;
        	      } else {
            	    CROW_LOG_WARNING << "unknown key:" << k << " value:" << v
                                 << " " << __FILE__ << " " << __func__ << " "
                                 << __LINE__;
				}
            }
            if (filename.size() < 5) {
              continue;  //	CROW_LOG_INFO << "skipping saving.";
            }
            CROW_LOG_INFO << "saving:" << filename;
            std::ofstream fd;
            fd.open("/tmp/" + filename, std::ios::out | std::ios::binary);
            fd << msg.parts[i].body;
            fd.close();
       }
   }
   return "it works!";
  });

  CROW_ROUTE(app, "/").methods(crow::HTTPMethod::GET) ([]() {
    CROW_LOG_INFO << "Hello";
    return "Hello world";
  });
  CROW_ROUTE(app, "/favicon.ico").methods(crow::HTTPMethod::GET) ([&](const crow::request& req, crow::response& res) {
    res.set_static_file_info("favicon.ico");
    res.end();
  });
  CROW_ROUTE(app, "/uploadForm").methods(crow::HTTPMethod::GET) ([&](const crow::request& req, crow::response& res) {
    res.set_static_file_info("uploadForm.html");
    res.end();
  });

/*
  CROW_ROUTE(app, "/json").methods(crow::HTTPMethod::GET)([&](){
  	crow::json::wvalue x({{"message", "Hello, World!"}});
  	x["message2"] = "Hello, World.. Again!";
  	return x;
	});
*/
  CROW_ROUTE(app, "/test").methods(crow::HTTPMethod::GET) ([&]() {
    nlohmann::json x;
    x["msg"] = "message";
    return x.dump();
  });

  // set the port, set the app to run on multiple threads, and run the app
  app.port(6780).multithreaded().run();
  //app.port(6780).ssl_file("../fullchain1.pem","../privkey1.pem").multithreaded().run();
  std::cout << "Crow quits.";
  return 0;
}

```
A strange problem occurs if the previous source code includes the following crow_all.h, the latest version.
```sh
$ ls -la ../crow_all.h
-rw-rw-r-- 1 a a 440956 Aug 27 15:43 ../crow_all.h
```
When the following command uploading a file,  
```sh
$ curl --verbose -F "user=1" -F file1=@Makefile -H "aa:b" http://localhost:6780/uploadFiles
```
The console of crow server shows the following messages.
```
(2022-11-04 10:32:25) [INFO    ] Crow/1.0 server is running at http://0.0.0.0:6780 using 6 threads
(2022-11-04 10:32:25) [INFO    ] Call `app.loglevel(crow::LogLevel::Warning)` to hide Info level logs.
(2022-11-04 10:32:25) [DEBUG   ] 0xaaaac1219e20 {0} queue length: 1
(2022-11-04 10:32:51) [DEBUG   ] 0xaaaac1219d40 {1} queue length: 1(2022-11-04 10:32:51) [DEBUG   ] 0xaaaac121c440 timer cancelled: 0xffffa3904178 0

(2022-11-04 10:32:51) [DEBUG   ] task_timer cancelled: 0xffffa3904178 0
(2022-11-04 10:32:51) [DEBUG   ] task_timer scheduled: 0xffffa3904178 1
(2022-11-04 10:32:51) [DEBUG   ] 0xaaaac121c440 timer added: 0xffffa3904178 1
(2022-11-04 10:32:51) [DEBUG   ] 0xaaaac121c440 timer cancelled: 0xffffa3904178 1
(2022-11-04 10:32:51) [DEBUG   ] task_timer cancelled: 0xffffa3904178 1
(2022-11-04 10:32:51) [DEBUG   ] task_timer scheduled: 0xffffa3904178 2
(2022-11-04 10:32:51) [DEBUG   ] 0xaaaac121c440 timer added: 0xffffa3904178 2
(2022-11-04 10:32:51) [DEBUG   ] 0xaaaac121c440 timer cancelled: 0xffffa3904178 2
(2022-11-04 10:32:51) [DEBUG   ] task_timer cancelled: 0xffffa3904178 2
(2022-11-04 10:32:51) [DEBUG   ] 0xaaaac121c440 from read(1) with description: "invalid HTTP method"
(2022-11-04 10:32:51) [DEBUG   ] 0xaaaac121c440 is_reading 0 is_writing 0
(2022-11-04 10:32:51) [DEBUG   ] 0xaaaac121c440 delete (idle) (queue length: 0)
(2022-11-04 10:32:51) [DEBUG   ] 0xaaaac121c440 timer cancelled: 0xffffa3904178 2
(2022-11-04 10:32:51) [DEBUG   ] task_timer cancelled: 0xffffa3904178 2
```
I mean "invalid HTTP method".
However, the problem disappears when including a previous version of crow_all.h. 
```sh
$ ls -la /ssd/crow_all.h 
-rw-rw-r-- 1 a a 440884 Jun 28 22:18 /ssd/crow_all.h
```
Is it possible to fix the problem for next version? Thanks a lot.

**Comment by kiner-shah** [comment]
Similar to #607 #626

**Comment by gittiver** [comment]
probably also solved with solution for #607 and #626
